niworkflows/viz/plots.py

```diff
@@ -222,32 +233,47 @@ def plot_carpet(
         from nilearn._utils import check_niimg_4d
         from nilearn._utils.niimg import _safe_get_data
 
+        # Load data
         img_nii = check_niimg_4d(
             img,
             dtype="auto",
         )
         func_data = _safe_get_data(img_nii, ensure_finite=True)
         func_data = func_data[..., nskip:]
         ntsteps = func_data.shape[-1]
-        data = func_data[atlaslabels > 0].reshape(-1, ntsteps)
-        oseg = atlaslabels[atlaslabels > 0].reshape(-1)
 
-        # Map segmentation
+        # Map segmentation to brain areas
+        # Boolean defining whether the default look up table is being used
+        default_lut = False
         if lut is None:
-            lut = np.zeros((256,), dtype="int")
+            default_lut = True
+            lut = np.zeros((261,), dtype="int")
             lut[1:11] = 1
             lut[255] = 2
             lut[30:99] = 3
             lut[100:201] = 4
+
+        if (lut < 0).any():
+            raise ValueError("The look up table should not contain negative values.")
+
         # Apply lookup table
-        seg = lut[oseg.astype(int)]
+        seg = lut[atlaslabels.astype(int)]
+
+        # Incorporate the crown in the carpetplot
+        seg[crown_mask] = seg.max() + 1
+
+        if (seg == 0).all():
+            raise ValueError("The segmented brain atlas is zero everywhere")
+
+        data = func_data[seg > 0].reshape(-1, ntsteps)
+        seg = seg[seg > 0].reshape(-1)
 
         # Decimate data
         data, seg = _decimate_data(data, seg, size)
         # Order following segmentation labels
         order = np.argsort(seg)[::-1]
         # Set colormap
-        cmap = ListedColormap(cm.get_cmap("tab10").colors[:4][::-1])
+        cmap = ListedColormap(cm.get_cmap("tab10").colors[: len(np.unique(seg))][::-1])
```

Comment:
This is the only change we want to keep -- having that hard-coded `4` is dangerous.